core/src/main/java/io/grpc/internal/ManagedChannelImpl.java

```diff
@@ -962,6 +962,18 @@ public void run() {
                   @Override
                   public void runInContext() {
                     setCall(newClientCall(method, callOptions));
+                    syncContext.executeLater(new Runnable() {
```

Comment:
Every call to executeLater _must_ be followed by a drain() call. I don't see such a call here or in callCancelled()

Reply:
The `IdleTimer` or `cancelIdleTimer()` will drain it. It only need be drained any time before IdleTimer is called.

Reply:
`isInUse()` is accessed in syncContext. So whenever you check `isInUse()`, it is drained. Otherwise it will eventually be drained by `IdleTimer` or `cancelIdleTimer()`

Reply:
IdleTimer: you are saying you are willing to wait for minutes, to hours, or maybe even forever (idle time could be infinite)? No, that's a bad idea.

isInUse(): That seems neither here nor there. It is called from idle mode logic, and we won't call exitIdleMode() unless we are in idle mode.

Even if those weren't broken, they are far too hard to verify and audit. Every call to executeLater needs to have a call to drain(). We use executeLater when we are processing within a lock, and then we call drain outside of the lock. I don't understand why we are using executeLater here.

Reply:
> I don't understand why we are using executeLater here.

My purpose of using `executeLater()` is to offload channel's tasks in syncCtx (could be a ton of them) from callExecutor thread. If it's not necessary performance wise, I will revert to `execute()`.

Reply:
The amount of processing in the Runnable is very small. I don't think we should be too concerned. In any case, we should favor correctness over performance so if this was important we'd need to figure out some other approach (which I do see some options available, but given the current inexpensive runnable it would be very hard to show significant improvements).

Reply:
Reverted.